src/expr/node_converter.h

```diff
@@ -0,0 +1,97 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Andrew Reynolds
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Node converter utility
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC4__EXPR__NODE_CONVERTER_H
+#define CVC4__EXPR__NODE_CONVERTER_H
+
+#include <iostream>
+#include <map>
+
+#include "expr/node.h"
+#include "expr/type_node.h"
+
+namespace cvc5 {
+
+/**
+ * A node converter for terms and types. Implements term/type traversals,
+ * calling the provided implementations of conversion methods (pre/postConvert
+ * and pre/postConvertType) at pre-traversal and post-traversal.
+ *
+ * This class can be used as a generic method for converting terms/types.
+ */
+class NodeConverter
```

Comment:
This has similar goals and structure (yet still different) to https://github.com/cvc5/cvc5/blob/704127085e5ba2deb19e41337908a340e1b191dd/src/expr/node_traversal.h#L40

Will it make sense to use that class? Or perhaps at least to document here when is it better to use this class than that class, or the differences? I think it might be undesirable to have the same functionality implemented twice without some explanations in the code.

Reply:
That is for node traversals, this is for node conversions.  The two are for different purposes, I'll mention this.